recipe_scrapers/grimgrains.py

```diff
@@ -0,0 +1,42 @@
+# mypy: allow-untyped-defs
+
+from itertools import chain
+from urllib.parse import urljoin
+
+from ._abstract import AbstractScraper
+from ._utils import get_minutes, get_yields
+
+
+class GrimGrains(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "grimgrains.com"
+
+    def author(self):
+        return "Hundred Rabbits"
+
+    def title(self):
+        return self.soup.main.h1.string
+
+    def total_time(self):
+        return get_minutes(self.soup.main.h2.string.split("—")[1])
+
+    def yields(self):
+        return get_yields(self.soup.main.h2.string.split("—")[0])
+
+    def image(self):
+        return urljoin(self.url, self.soup.main.find("img")["src"])
+
+    def ingredients(self):
+        return [
+            f"{i.b.text}: {i.u.text}"
+            for i in chain.from_iterable(self.soup.main("dl", "ingredients"))
+            if i.b
+        ]
+
+    def instructions(self):
+        instructions = [i("li") for i in self.soup.main("ul", "instructions")]
+        return "\n".join([i.text for i in chain.from_iterable(instructions)]) or None
```

Comment:
Sorry about the back-and-forth on this: I think I've decided that this `or None` fallback isn't necessary.  A recipe _should have_ at least one instruction; if it doesn't, then I'm not sure it's a recipe.  I'm going to do a bit of cleanup in the codebase for that too.